packages/mobile/src/geth/saga.ts

```diff
@@ -98,6 +100,21 @@ export function* waitForGethConnectivity() {
   }
 }
 
+export async function waitForGethConnectivityAsync() {
+  let connected = isGethConnectedSelector(store.getState())
```

Comment:
We may want to wait for the node to be synced rather than simply connected.

An example of when this distinction is relevant is if the user has the app in the background for 5+ minutes, then opens the app again and immediately makes a transaction. In this case, the node will become connected again quickly, but it will be at least 60 blocks behind the head. If the transaction is sent immediately, then sync occurs, an error will be reported that the transaction "was not mined within 50 blocks" because the chain head just "jumped" 60 blocks.

There isn't a ready-made redux state entry for this, but the [chain head selector](https://github.com/celo-org/celo-monorepo/blob/94f47c0fa17308273ce30df86964cc88a0f409bb/packages/mobile/src/geth/selectors.ts#L9) provides the data needed, and the [sync status saga](https://github.com/celo-org/celo-monorepo/blob/94f47c0fa17308273ce30df86964cc88a0f409bb/packages/mobile/src/geth/saga.ts#L302) provides an example of how it can be used.

Reply:
What's a realistic upper bound for sync time?